### ISSUE TITLE
"no matter"

### ISSUE THREAD

**Issue by nschneid** [UniversalDependencies/UD_English-EWT#376]
In [EWT](http://universal.grew.fr/?custom=6354d17e08c23) it is currently: advmod(matter/ADV/RB, no/ADV/RB), which is counterintuitive to me.

For the internal structure, GUM uses a compositional analysis: det(matter/NOUN/NN, no/DET/DT)

In PTB guidelines I don't see any discussion of the tags or the internal structure of the expression. There is an example where "no matter" forms an ADVP with an SBAR complement. CGEL p. 761 says "no matter" is a single lexeme that takes a subordinate clause complement, and the bottom of p. 764 claims that the complement can be reduced. Simple English Wikipedia (based on CGEL's POS inventory) classifies it [as a preposition](https://simple.wiktionary.org/wiki/no_matter) (analogous to the PTB `IN` tag). [Rawlins (2008, p. 80)](https://rawlins.io/downloads/rawlins_dissertation_2008.pdf) tentatively suggests "no matter" is a complementizer. [Jong-Bok and Han (2011)](https://oak.go.kr/central/journallist/journaldetail.do?article_seq=18348) seem to treat it as a noun expression that takes a complement.

This leaves open the possibility of treating "no matter" as a marker, which suggests it should be connected via `fixed` rather than `det`. For "no matter what it costs", it would be `mark`; and `case` seems natural for "no matter the cost". But if we treat it as formed via `det`, the case for attaching it as `mark` or `case` becomes weaker.

Or, it could be treated as a nominal formed by `det` that takes an `acl` complement. This works for "no matter what it costs" (or "no matter what", which can be considered to contain a reduced clause). But is `acl` appropriate for "no matter the cost"? Should we say that is `obj` (analogous to worth/ADJ having an object)? Or `nmod:npmod`, perhaps?

In any case, I don't think this expression licenses a relative clause, which is annotated in some places in EWT. I will implement the following for now:

- For the internal structure, follow GUM and treat the expression as transparent: det(matter/NOUN/NN, no/DET/DT)
- Treat what follows as a dependent with `acl` (if a clause or expression suggestive of a reduced clause) or `nmod:npmod` (if it has a nominal complement which does not look like ellipsis).
- For the attachment of the full "no matter..." phrase (headed by "matter"): `obl:npmod`

**Comment by nschneid** [comment]
Kim and Lee argue that the internal structure should be regarded as fixed because it is not productive: *_little matter_, *_no matters_. AFAICT all the above sources treat it as a fixed expression. I think `fixed` would be warranted on this basis, with the marker (`mark`/`case`) analysis paralleling several other expressions that we treat as `fixed`.

**Comment by nschneid** [comment]
"no matter the cost" with nsubj like 'the cost is no matter': I see the logic semantically but it looks totally weird to have a nominal predicate preceding its subject with no copula. I would rather have a noun with an object than this.

It's not obvious to me that "no matter the cost" is a clause...seems not too far from "apart from the cost" or "regardless of the cost".

**Comment by amir-zeldes** [comment]
I mean, if there's a predication in there then I think it should be a clause. If it's not a subject, then how about the advcl analysis? We need to be able to explain what's happening with "no matter if/whether" etc.

**Comment by nschneid** [comment]
I'm not convinced "no matter the cost" has a predication:

1. Clauses can usually be negated, right? Even verbless clauses like "With his feet on the table, ..." can be negated to "With his feet not on the table". But you can't say "no matter not the cost" or "no matter no cost" or remove the "no".
2. How is it more predication-like than "regardless of the cost"?

The fixed+case/mark analysis seems simplest to me—that way we say it can mark a clause or an NP. But I'd like to hear opinions from others.

**Comment by amir-zeldes** [comment]
> Clauses can usually be negated, right? 

No, not necessarily. Some clauses are always and only negative ("lest" clauses come to mind), and some are only positive (comparative corelatives, you can't say "*the not faster the better"). Some languages systematically use different clause types based on the positive-negative axis (Coptic is notable in having lots of these).

> How is it more predication-like than "regardless of the cost"?

Because you can say "no matter if they come or not". "Regardless" requires mediation with "of". That's not necessarily a deciding factor, I just find the pattern a bit odd for a standard 'preposition'. If it only took nominals I guess I could see it as equivalent to "regardless" (though really I think that's an adverb which takes a PP complement headed by "of"). But what would you do with the "no matter whether" cases? Would you want double mark here?

**Comment by nschneid** [comment]
"Lest" clauses can be negated: "Lest we not arrive on time, ..."

But you're right about comparative correlatives. Those are one of the minor clause types discussed in CGEL pp. 944-945.

My point about "regardless of" was not to make an argument about the POS of "regardless", just to say that if it's not considered a predication, then I'm not sure why "no matter the cost" should be considered a predication.

UD does allow P + "whether" clauses as advcl: "They disagree about whether..." etc. I think those are properly considered complements (the whole cobl discussion), but formally speaking, there would be nothing exceptional about double mark on an advcl.

As true adjuncts, there are also "whether or not..." and "whether X or Y" type clauses, in case that is relevant.

So I guess our options are

- decide that "no matter X" is a minor clause type—the only *adjunct* type that prohibits negation to my knowledge—that does not begin with a marker, or
- decide that "no matter" has grammaticalized as a marker, and can function as `case` or `mark`, including before "whether".

**Comment by amir-zeldes** [comment]
Huh, I only find "lest we not" followed by "forget", and discussions about whether that's correct (or whether that even has double negative meaning):

https://www.quora.com/What-does-lest-we-forget-mean-1
https://groups.google.com/g/alt.usage.english/c/3IDMCNlDc0I?pli=1

I think there are some theoretical papers about constructions like these which allow inserting a negation without a change in meaning. 

I think there are a bunch of constructions that come with built in negation but are still clauses, like "not to brag, but..." where you can't do a positive "??to brag, ...". Of course, this shows some degree of fossilization, but I have taken the UD policy to be preferring a compositional analysis whenever possible, including within names and other fossilized expressions. And in languages other than English, asymmetry between positive and negative clause possibilities is sometimes much greater, so either way I wouldn't take negation as an indication of clause status.

As for the options, TBH I'm not categorically opposed to a case or mark analysis, but I'm just wondering if it's really necessary. Essentially I think the list of 'fixed' in English is getting bigger than it should be, so I'm a bit loathe to see it grow. The internal structure of "no matter" seems pretty normal to me (similar to "no reason to fret" and others), so I'd like to see the 'det' relation survive (also fits the established xpos).

**Comment by nschneid** [comment]
I mean "lest" is largely archaic to begin with. But I see no reason to believe "lest" clauses resist negation. Examples from the web:

- **Lest the reader not understand**, Bailey specified the sort of “small dogmatisms” that ought to be put away.
- We see this principle reflected also in restrictions against workers moonlighting in second jobs, **lest they not have** the energy to devote themselves fully to their first job
- I'm always a bit disinclined to change config files for things like, **lest they not get changed back**

"Not to brag" is just a normal negated infinitival clause with an idiomatic meaning. I think the same syntactic construction as "To elaborate, ...".

**Comment by amir-zeldes** [comment]
> I see no reason to believe "lest" clauses resist negation

OK, I'm convinced about lest - but there are definitely clauses that can only appear in the positive/negative, even though it's very rare in English (in Coptic it's systematic for certain tenses/aspects).

> just a normal negated infinitival clause 

I think syntactically that's right, but I also think "no matter" is just a normal negated noun phrase like "no reason". They both share the property that you can't just un-negate them ("*to brag, ..." just doesn't work/mean what the idiom means). That doesn't mean that either of them is totally opaque or unstructured, and there are variants like "not to brag too much", or for "matter" we can find:

  * Little matter that ninety percent of the money he earns comes from framing pictures (Paul Auster)

Or similar expressions with another noun instead of 'matter', such as 'wonder':

  * No wonder you're hungry
  * Little wonder that he quotes with relish from David Lodge's hilarious academic novels
  * Small wonder that you feel as bad as you do this morning

I just don't want all of these to be a huge list of fixed expressions - they are certainly collocations from a frequentist perspective, but I think UD's job is just to describe the syntactic structure, which is a noun head with a negation. I'm less passionate about it being an advcl, or having a subj, I'm just trying to avoid a long shopping list of fixed, and especially when I think the internal structure is not headless.

**Comment by nschneid** [comment]
I absolutely would not want to use `fixed` for the "little matter"/"no wonder" ones. That appears to use a lexically more productive construction, and I'd say the noun heads the clause. It is perhaps historically related to "no matter whether it costs too much", but that and "no matter the cost" can only function as adjuncts, not as main clauses. Also:

- *No wonder the cost!
- *Little matter the cost.

**Comment by amir-zeldes** [comment]
OK, but is there any analysis you'd be OK with that keeps "matter" as the head of "no", so we can keep the normal structure there? Otherwise we're pretending it's not a negation, which I think it still is.

**Comment by nschneid** [comment]
That depends on the structure IMO. If we treat "no matter" as a grammaticalized marker, then I think fixed makes the most sense. We're saying the historical determiner relationship is no longer productive in that construction. We can still keep DET and NOUN as tags.

But I want to hear others' opinions.

**Comment by amir-zeldes** [comment]
> If we treat "no matter" as a grammaticalized marker, then I think fixed makes the most sense

Yeah, I'm the least comfortable with this part - "no" is still acting as a normal negation here, meaning that it doesn't matter. So I think "matter" should be the head.

> I want to hear others' opinions.

Me too - @dan-zeman @manning , any thoughts?

**Comment by dan-zeman** [comment]
I would do det(matter/NOUN, no/DET). That is, the structure attributed to GUM in the first post in this thread.

And I think I would use obl(superordinate-clause, matter)... although advcl seems justifiable, too.

**Comment by nschneid** [comment]
@dan-zeman What would you do for the relation between "matter" and "cost"?

**Comment by dan-zeman** [comment]
> @dan-zeman What would you do for the relation between "matter" and "cost"?

`nmod(matter, cost)`

**Comment by amir-zeldes** [comment]
The English guidelines reserve nmod (and obl) to cases where there is an explicit preposition as a case marker, so that would introduce an inconsistency. We could go with nmod:npmod, but if we view the 'no matter' as advcl, then I would argue it should be obl:npmod as a clause modifier (since it doesn't actually belong to an NP headed by 'matter'). This is also one of the options I suggested above. 

In sum, I think either obl:npmod or nsubj (if we view this as an argument structure like "it doesn't matter that X") are the best options.

**Comment by nschneid** [comment]
Yes, as @amir-zeldes points out we use the `:npmod` subtype for non-prepositional nmod/obl. Given that I gather @dan-zeman is arguing for

det(matter, no)
obl:npmod(PRED, matter)
nmod:npmod(matter, cost) in "no matter the cost"
acl(matter, costs) in "no matter whether it costs too much"

I think this is the least objectionable "shallow" analysis, i.e. the analysis that treats "no" and "matter" as literally as possible in their compositionality. But it is not ideal: a noun licensing either acl or nmod:npmod seems exceptional for the language (most nouns that license complements would have PP complements if nominal, thus nmod rather than nmod:npmod—e.g. "the fact of" vs. "the fact that").

My native speaker intuition is that "no matter" really functions as a marker (case/mark) that happens to be spelled as two words for historical reasons, but is effectively a single lexeme with build-in negation semantics. I.e. it is a single word with negation in the same way as "nobody", "without", or "notwithstanding". It happens to be conventionally spelled as two words, but I wouldn't be surprised to find people writing "nomatter", and in fact this is [attested on Twitter](https://twitter.com/search?q=nomatter&src=typed_query).

Note that we decided to allow some VERBs as case/mark because they have become similarly grammaticalized.

But I do see the practical appeal of shallowly recognizing the "det" in no+matter as it would be if "no matter" were used in other contexts ("It is no matter", etc.).

**Comment by nschneid** [comment]
P.S. No matter what we decide :), I predict annotators will have to look it up on the [fixed page](https://universaldependencies.org/en/dep/fixed.html) (which is essentially a mini-constructicon).
